apps/server/src/toasts/toast.service.ts

```diff
@@ -2,15 +2,173 @@ import { Injectable } from '@nestjs/common';
 import { Toast } from './entities/toast.entities';
 import { InjectModel } from '@nestjs/sequelize';
 import { CreateToastDto } from './dto/create-toast.dto';
+import { Op, Sequelize } from 'sequelize';
 
 @Injectable()
 export class ToastService {
-  constructor(@InjectModel(Toast) private toastModel: typeof Toast) {}
+  constructor(@InjectModel(Toast) public toastModel: typeof Toast) {}
+
   findAll() {
     return this.toastModel.findAll();
   }
+  getPastToastForUser(userId: string) {
+    const currentDate = new Date();
+    return this.toastModel.findAll({
+      where: { userId, toastDate: { [Op.lt]: currentDate } },
+    });
+  }
+
+  getAllPastToasts() {
+    const currentDate = new Date();
+
+    const allPast = this.toastModel.findAll({
+      where: { toastDate: { [Op.lt]: currentDate } },
+    });
+    console.log(currentDate);
+    return allPast;
+  }
+
+  getAllFutureToasts() {
+    const currentDate = new Date();
+    return this.toastModel.findAll({
+      where: { toastDate: { [Op.gt]: currentDate } },
+    });
+  }
+
+  getAmountToastsForCurrentPeriod() {
+    const currentDate = new Date();
+    const firstDateForPeriod = new Date(currentDate.getFullYear(), 0, 1);
```

Comment:
not sure let's talk about it tomorrow , if I leave it as static const then it will create a new date when I run the server , instead of when I run the function....
not sure if it matters though